main.go

```diff
@@ -134,6 +134,12 @@ func main() {
 			LeaderElection:         enableLeaderElection,
 			LeaderElectionID:       "a4ddbf36.kubewarden.io",
 			ClientDisableCacheFor:  []client.Object{&corev1.ConfigMap{}, &appsv1.Deployment{}},
+			// Watch for kubewarden namespace on namespaced resources:
```

Comment:
(randomly opening thread here)

So, I think this change is succint. It may be not transparent enough, I suppose. But we should never need to look outside of our namespace, besides for cluster-scoped resources.

Another option is to configure the Manager being used in `SetupWithManager()` , for both the clusteradmissionpolicy_controller.go and admissionpolicy_controller.go. E.g, here:

https://github.com/kubewarden/kubewarden-controller/blob/50d6c05f92f85e68bc50c5403b70aa0cbbbc2a3f/controllers/clusteradmissionpolicy_controller.go#L69-L73

Something like:
```go
namespace := deploymentsNamespace;
options := ctrl.Options{
    Namespace: cache.MultiNamespacedCacheBuilder(strings.Split(namespace, ","))
}
mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), options)
// and then keep using it:
err := ctrl.NewControllerManagedBy(mgr)
(...)
```

In addition, it may be possible to move away from using a broad `Watches()`, and filter for Pods with out own labels. This would improve efficiency on clusters where there's a lot of Pods. But it shouldn't be an issue anymore, as we will be looking only into the kubewarden namespace anyways.